lease/leasehttp/http.go

```diff
@@ -188,9 +188,14 @@ func TimeToLiveHTTP(ctx context.Context, id lease.LeaseID, keys bool, url string
 			errc <- err
 			return
 		}
-		b, err = ioutil.ReadAll(resp.Body)
-		resp.Body.Close()
+		b, err = readFunc(resp)
 		if err != nil {
+			// follower cannot reach leader at the moment
+			// possibly from network errors
+			// translate to ErrNoLeader, expecting clients to retry
+			if err == io.EOF || err == io.ErrUnexpectedEOF {
+				err = rpctypes.ErrNoLeader
```

Comment:
should we return grpc error `ErrGRPCNoLeader` instead of client side error `rpctypes.ErrNoLeader`